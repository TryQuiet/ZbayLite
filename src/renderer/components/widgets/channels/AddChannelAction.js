import React from 'react'
import * as R from 'ramda'

import IconButton from '@material-ui/core/IconButton'
import { withStyles } from '@material-ui/core/styles'

import addIcon from '../../../static/images/zcash/add-icon.svg'
import MenuAction from '../../ui/MenuAction/MenuAction'
import MenuActionItem from '../../ui/MenuAction/MenuActionItem'
import CreateChannelModal from '../../../containers/widgets/channels/CreateChannelModal'

const styles = theme => ({
  icon: {
    width: 15,
    height: 15
  },
  button: {
    width: 25,
    height: 25,
    padding: 4,
    color: theme.typography.body2.color,
    backgroundColor: 'rgb(0,0,0,0.26)',
    borderRadius: '50%'
  }
})

export const AddChannelAction = ({ classes, openCreateModal }) => {
  return (
    <React.Fragment>
      <MenuAction
        classes={{
          button: classes.button,
          icon: classes.icon
        }}
        icon={addIcon}
        iconHover={addIcon}
        IconButton={IconButton}
        offset='0 8'
      >
        <MenuActionItem onClick={openCreateModal} title='Create' />
      </MenuAction>
      <CreateChannelModal />
    </React.Fragment>
  )
}

export default R.compose(
  React.memo,
  withStyles(styles)
)(AddChannelAction)
