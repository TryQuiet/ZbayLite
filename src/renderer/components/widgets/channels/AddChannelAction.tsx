import React from 'react'

import IconButton from '@material-ui/core/IconButton'
import { makeStyles } from '@material-ui/core/styles'

import addIcon from '../../../static/images/zcash/add-icon.svg'
import MenuAction from '../../ui/MenuAction'
import MenuActionItem from '../../ui/MenuActionItem'
import CreateChannelModal from '../../../containers/widgets/channels/CreateChannelModal'
import { Action, ActionFunction0 } from 'redux-actions'

const useStyles = makeStyles((theme) => ({
  icon: {
    width: 15,
    height: 15
  },
  button: {
    width: 25,
    height: 25,
    padding: 4,
    color: theme.typography.body2.color,
    backgroundColor: 'rgb(0,0,0,0.26)',
    borderRadius: '50%'
  }
}))

type AddChannelActionProps = {
  openCreateModal: ActionFunction0<Action<{
    modalName: string;
    data: any;
  }>>
}

export const AddChannelAction: React.FC<AddChannelActionProps> = ({ openCreateModal }) => {
  const classes = useStyles({})

  return (
    <React.Fragment>
      <MenuAction
        classes={{
          button: classes.button,
          icon: classes.icon
        }}
        icon={addIcon}
        iconHover={addIcon}
        IconButton={IconButton}
        offset='0 8'
      >
        <MenuActionItem onClick={openCreateModal} title='Create' />
      </MenuAction>
      <CreateChannelModal />
    </React.Fragment>
  )
}

export default AddChannelAction
