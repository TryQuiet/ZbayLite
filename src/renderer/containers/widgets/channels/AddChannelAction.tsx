import React from 'react'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import * as R from 'ramda'

import AddChannelAction from '../../../components/widgets/channels/AddChannelAction'
import { actionCreators } from '../../../store/handlers/modals'

export const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      openCreateModal: actionCreators.openModal('createChannel')
    },
    dispatch
  )

export default R.compose(
  React.memo,
  connect(
    null,
    mapDispatchToProps
  )
)(AddChannelAction)
