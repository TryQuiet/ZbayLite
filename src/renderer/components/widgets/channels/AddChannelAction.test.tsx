import React from 'react'
import { shallow } from 'enzyme'

import { AddChannelAction } from './AddChannelAction'

describe('BaseChannelsList', () => {
  // TODO: [refactoring] test useState when enzyme is up to date
  it('renders component', () => {
    const openModal = jest.fn()
    const result = shallow(<AddChannelAction openCreateModal={openModal} />)
    expect(result).toMatchSnapshot()
  })
})
