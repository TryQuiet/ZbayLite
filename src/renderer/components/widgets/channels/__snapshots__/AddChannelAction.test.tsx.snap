// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseChannelsList renders component 1`] = `
<Fragment>
  <Component
    icon="test-file-stub"
    iconHover="test-file-stub"
    offset="0 8"
  >
    <Component
      onClick={[MockFunction]}
      title="Create"
    />
  </Component>
  <Memo(withModal(Component)) />
</Fragment>
`;
